lib/identity_cache/configuration_dsl.rb

```diff
@@ -205,11 +205,20 @@ def build_recursive_association_cache(association, options) #:nodoc:
         options[:cached_accessor_name]   = "fetch_#{association}"
         options[:records_variable_name]  = "cached_#{association}"
 
-        self.class_eval(<<-CODE, __FILE__, __LINE__ + 1)
-          def #{options[:cached_accessor_name]}
-            fetch_recursively_cached_association('#{options[:records_variable_name]}', :#{association})
-          end
-        CODE
+        if method_defined?(options[:cached_accessor_name].to_sym)
+          self.class_eval(<<-CODE, __FILE__, __LINE__ + 1)
+            def #{options[:cached_accessor_name]}_with_idc
```

Comment:
Thank you, comment. :smile: 

I understand define fetch_polymorphic_records. 
I red

- ActiveRecordObjects
- #build_recursive_association_cache
- #build_id_embedded_has_many_cache

:eyes: 
--

Thats is certainly redefin #fetch_polymorphic_records in SchemaChangeTest#test_embed_existing_cache_has_many.